src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -578,15 +591,16 @@ private async Task LoopProducerWorker()
                 try
                 {
                     int circularIndex = 0;
-                    for (; ; )
+                    while (await _toProduce.OutputAvailableAsync())
```

Comment:
I'm not sure what policy ML.NET has used for async in general, but generally we recommend libraries use `.ConfigureAwait(false)` on all awaits.

Reply:
Good call. This was a mistake.